nilearn/_utils/compat.py

```diff
@@ -43,3 +49,8 @@ class _urllib():
         request = module_lookup
         error = module_lookup
         parse = module_lookup
+
+    def md5_hash(string):
```

Comment:
This is doable although I find more clear to have a separation between python2 and python3 code.